lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2273,6 +2282,38 @@ def wait_for_vm_ip(self, vm, poll=100, sleep=5):
 
         return facts
 
+    def get_vm_events(self, eventTypeIdList):
+        newvm = self.get_vm()
+        byEntity = vim.event.EventFilterSpec.ByEntity(entity=newvm, recursion="self")
+        filterSpec = vim.event.EventFilterSpec(entity=byEntity, eventTypeId=eventTypeIdList)
+        eventManager = self.content.eventManager
+        return eventManager.QueryEvent(filterSpec)
+
+    def wait_for_customization(self, vm, poll=10000, sleep=10):
+        facts = {}
+        thispoll = 0
+        while thispoll <= poll:
+            eventStarted = self.get_vm_events(['CustomizationStartedEvent'])
+            if len(eventStarted):
+                thispoll = 0
+                while thispoll <= poll:
+                    eventsFinishedFailed = self.get_vm_events(['CustomizationSucceeded', 'CustomizationFailed'])
+                    if len(eventsFinishedFailed):
+                        if not isinstance(eventsFinishedFailed[0], vim.event.CustomizationSucceeded):
+                            raise RuntimeError('Customization failed with error {0}:\n{1}'.format(
```

Comment:
Never raise an exception, fail the module instead

Reply:
Changed it, but how do I properly return the error message? By using `self.module.fail_json`?

Reply:
Yes